packages/plugins/typescript/effect/src/visitor.ts

```diff
@@ -0,0 +1,198 @@
+import autoBind from 'auto-bind';
+import { type GraphQLSchema, type OperationDefinitionNode, print } from 'graphql';
+import {
+  type ClientSideBasePluginConfig,
+  ClientSideBaseVisitor,
+  DocumentMode,
+  LoadedFragment,
+  type RawClientSideBasePluginConfig,
+} from '@graphql-codegen/visitor-plugin-common';
+
+export interface EffectPluginConfig extends ClientSideBasePluginConfig {}
+
+const additionalStaticContent = (documentMode: DocumentMode) => `
+export type GraphQLSuccessResponse<A> = {
+  body: ExecutionResult & { data: A };
+  headers: Record<string, string>;
+};
+
+export type GraphQLErrorResponse = {
+  body: Omit<ExecutionResult, 'data'>;
+  headers: Record<string, string>;
+};
+
+export class MissingDataGraphQLResponseError extends Data.TaggedError(
+  'MissingDataGraphQLResponseError',
+)<GraphQLErrorResponse> {}
+
+export type GraphQLOperationOptions = {
+  preferredOpName?: string;
+};
+
+type GraphQLOperationArgs = {
+  document: ${documentMode === DocumentMode.string ? 'string' : 'DocumentNode'};
+  fallbackOperationName: string;
+};
+
+// https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#legacy-watershed
+const Accept = 'application/graphql-response+json; charset=utf-8, application/json; charset=utf-8';
+
+const makeGraphQLClient = Http.client.mapRequest(
+  Http.request.setHeaders({ Accept, 'Content-Type': 'application/json' }),
+);
+
+const makeGraphQLOperation =
+  <Vars, Data>({ document, fallbackOperationName }: GraphQLOperationArgs) =>
+  (variables: Vars, opts?: GraphQLOperationOptions) => {
+    const operationName = opts?.preferredOpName ?? fallbackOperationName;
+    const query = ${documentMode === DocumentMode.string ? 'document' : 'print(document)'};
+
+    return Effect.flatMap(Http.client.Client, client =>
+      Http.request.post('').pipe(
```

Comment:
I wonder if creating a `GraphQLConfig` tag would be better, and storing the endpoint in there? Ensuring it is unique might be an issue.

Reply:
I was thinking about the base URL and the endpoint configuration quite a bit. It is customary to have a single GraphQL endpoint per schema. However, I cannot be certain that the generated SDK is meant for a single schema only. For example:

> You can also specify `schema` on both levels: root and output-file, and then GraphQL Code Generator will merge both schemas into one

https://the-guild.dev/graphql/codegen/docs/config-reference/schema-field#multiple-schemas-and-client-side-schema

It could be that the generated SDK combines multiple schemas. Therefore, different operations may need to be executed against different GraphQL endpoints. So, I wanted to avoid making assumptions and let the users specify their base clients when they provide the Http Client layer. The consuming code could manage as many clients as they need and handle the mapping of clients to operations on their end.